fairworkflows/rdf_wrapper.py

```diff
@@ -0,0 +1,54 @@
+import warnings
+
+import rdflib
+
+
+class RdfWrapper:
+    def __init__(self, uri):
+        self._rdf = rdflib.Graph()
+        self._uri = uri
+        self.this = rdflib.URIRef(self._uri)
```

Comment:
I am not fully convinced with the naming of `self.this`. Do you have any good ideas? Maybe `self.self_ref`, `self.rdf_uri`, `self.uri_rdf`?

Reply:
Yes, I agree and have also been wondering what to call this. I'm leaning towards ```self.self_ref```.

The more difficult issue, that this overlaps with, is the problem of using the URI to uniquely identify a step. This works if that URI is to a published step (i.e. nanopublished) but when the steps are local or unpublished, they just have this 'default' URI. That is a significant issue that needs resolved. I was thinking the best fix for that might be to have a local "cache/" dir in which we can store 'unpublished' steps. The file name would be chosen using e.g. tempfile to get a unique file name. When published, we'd need to then somehow update that URI, including in any rdf that references it. This could be messy when dealing with one or more ```FairWorkflow``` objects that reference unpublished steps. I'm wondering if instead we should be using rdflib 'blank nodes' or something instead. Or ```FairStep``` needs an ```isLocal()``` or ```isPublished()``` method that the FairWorkflow would check to update the URIs in its rdf graph.

Reply:
I changed it to `self.self_ref`. 

Good ideas on problem of non-unique URIs. Let's dicsuss what is the best way! (and fix that in a different issue)